File: frontend/admin/src/types/notification.ts
export interface Notification {
    id: string
    title: string
    message: string
    user_id: string
    is_read: boolean
    tenant_id: string
    created_at: string
    updated_at: string
}

export interface NotificationCreate {
    title: string
    message: string
    send_to_all?: boolean
    user_ids?: string[]
    unit_ids?: string[]
}

export interface NotificationSendResult {
    success: boolean
    count: number
    notifications: Notification[]
}

File: backend/app/models/notification.py
"""Notification domain model."""
from sqlalchemy import Column, String, Boolean, DateTime, ForeignKey, func
from sqlalchemy.orm import relationship

from app.core.database import Base
from app.core.uuid import generate_uuid


class Notification(Base):
    __tablename__ = "notifications"

    id = Column(String, primary_key=True, default=generate_uuid)
    title = Column(String, nullable=False)
    message = Column(String, nullable=False)
    is_read = Column(Boolean, default=False)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    scheduled_for = Column(DateTime(timezone=True), nullable=True)
    sent_at = Column(DateTime(timezone=True), nullable=True)

    user_id = Column(String, ForeignKey("users.id"), nullable=False)
    user = relationship("User")

    tenant_id = Column(String, ForeignKey("tenants.id"), nullable=False)
    tenant = relationship("Tenant")
